src/main/kotlin/org/opensearch/replication/seqno/RemoteClusterRetentionLeaseHelper.kt

```diff
@@ -60,15 +73,60 @@ class RemoteClusterRetentionLeaseHelper constructor(val followerClusterName: Str
             return true
         }
         catch (e: RetentionLeaseNotFoundException) {
+            return addNewRetentionLeaseIfOldExists(leaderShardId, followerShardId, seqNo)
+        }catch (e : Exception) {
             return false
         }
         return true
     }
 
+    private suspend fun addNewRetentionLeaseIfOldExists(leaderShardId: ShardId, followerShardId: ShardId, seqNo: Long): Boolean {
+        //Check for old retention lease id
+        val oldRetentionLeaseId = retentionLeaseIdForShard(followerClusterName, followerShardId)
+        val requestForOldId = RetentionLeaseActions.RenewRequest(leaderShardId, oldRetentionLeaseId, seqNo, retentionLeaseSource)
+        try {
+            client.suspendExecute(RetentionLeaseActions.Renew.INSTANCE, requestForOldId)
```

Comment:
Shouldn't we use similar mechanism as `verifyRetentionLeaseExist` to understand if old one exists?

Reply:
Yes we are doing exactly the same thing, we are verifying presence of old retention lease if by trying to renew it.
If the renewal is successful, we add new retention lease, if not we do not add new retention lease.

Reply:
it is better to re-use the method but even then in new portion of the code, we are renewing with actual seqno here rather than `RETAIN_ALL` (as in verifyRetentionLeaseExist). Could you cross check again?

Reply:
The value of seq number is sent by leader as RETAIL_ALL only
https://github.com/opensearch-project/cross-cluster-replication/blob/main/src/main/kotlin/org/opensearch/replication/repository/RemoteClusterRestoreLeaderService.kt#L114